File: extract.js
const fs = require('fs');
const pdf = require('pdf-parse');

console.log(typeof pdf);

let dataBuffer = fs.readFileSync('public/INDO 100.pdf');

if (typeof pdf === 'function') {
  pdf(dataBuffer).then(function(data) {
      fs.writeFileSync('extracted_text.txt', data.text);
      console.log("Extraction complete.");
  }).catch(console.error);
} else if (pdf.default && typeof pdf.default === 'function') {
  pdf.default(dataBuffer).then(function(data) {
      fs.writeFileSync('extracted_text.txt', data.text);
      console.log("Extraction complete.");
  }).catch(console.error);
} else {
  console.log('Unable to find pdf parse function:', pdf);
}

File: extract.py
import sys
try:
    import pypdf
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "pypdf"])
    import pypdf

reader = pypdf.PdfReader("public/INDO 100.pdf")
text = ""
for page in reader.pages:
    text += page.extract_text() + "\n"

with open("extracted_text.txt", "w", encoding="utf-8") as f:
    f.write(text)
print("Done")
